components/script/dom/globalscope.rs

```diff
@@ -1241,18 +1246,52 @@ impl GlobalScope {
     }
 
     fn read_file(&self, id: Uuid) -> Result<Vec<u8>, ()> {
+        let recv = self.send_msg(id);
+        GlobalScope::read_msg(recv)
+    }
+
+    pub fn read_file_async(
+        &self,
+        id: Uuid,
+        promise: Rc<Promise>,
+        f: Box<dyn Fn(Rc<Promise>, Result<Vec<u8>, ()>) + Send>,
+    ) {
+        let recv = self.send_msg(id);
+        let trusted_promise = TrustedPromise::new(promise);
+        let canceller = self.task_canceller(TaskSourceName::FileReading);
+        let task_source = self.file_reading_task_source();
+        thread::spawn(move || {
```

Comment:
I think you can use an IPC route on `recv` instead of a thread, see for example https://github.com/servo/servo/blob/6d5d350350e7550927b5c214460b0c1fb89d114f/components/script/dom/globalscope.rs#L472

Reply:
I am facing some problem in this part. I have made the above changes to some extent. I am facing a couple of errors which I could not figure out how to get past.
```
error[E0507]: cannot move out of `trusted_promise`, a captured variable in an `FnMut` closure
    --> components/script/task.rs:27:15
     |
27   |           $name(move || $body)
     |                 ^^^^^^^ move out of `trusted_promise` occurs here
     | 
    ::: components/script/dom/globalscope.rs:1259:13
     |
1259 |           let trusted_promise = TrustedPromise::new(promise);
     |               --------------- captured outer variable
...
1279 | /                             task!(resolve_promise: move || {
1280 | |                                 let promise = trusted_promise.root();
     | |                                               ---------------
     | |                                               |
     | |                                               move occurs because `trusted_promise` has type `dom::bindings::refcounted::TrustedPromise`, which does not implement the `Copy` trait
     | |                                               move occurs due to use in closure
1281 | |                                 let _ac = enter_realm(&*promise.global());
1282 | |                                 f(promise, bytes);
1283 | |                             }),
     | |______________________________- in this macro invocation

error[E0507]: cannot move out of `f`, a captured variable in an `FnMut` closure
    --> components/script/task.rs:27:15
     |
27   |           $name(move || $body)
     |                 ^^^^^^^ move out of `f` occurs here
     | 
    ::: components/script/dom/globalscope.rs:1256:9
     |
1256 |           f: Box<dyn Fn(Rc<Promise>, Result<Vec<u8>, Error>) + Send>,
     |           - captured outer variable
...
1279 | /                             task!(resolve_promise: move || {
1280 | |                                 let promise = trusted_promise.root();
1281 | |                                 let _ac = enter_realm(&*promise.global());
1282 | |                                 f(promise, bytes);
     | |                                 -
     | |                                 |
     | |                                 move occurs because `f` has type `std::boxed::Box<dyn std::ops::Fn(std::rc::Rc<dom::promise::Promise>, std::result::Result<std::vec::Vec<u8>, dom::bindings::error::Error>) + std::marker::Send>`, which does not implement the `Copy` trait
     | |                                 move occurs due to use in closure
1283 | |                             }),
     | |______________________________- in this macro invocation

```
I would need some help here. I have pushed the changes (for you to review). Hopefully it is close to what you had asked for :)